### ISSUE TITLE
Don't truncate a string in the error log

### ISSUE THREAD

**Issue by jostephd** [wesnoth/wesnoth#3876]
[This post](https://forums.wesnoth.org/viewtopic.php?p=637604#p637580) includes the following error log:

```
20190116 15:22:17 error scripting/lua: In function lua_kernel::run(): Lua Error:
 When executing, Lua runtime error: [string " wesnoth.dofile("~add-ons/Legend_of_the_Invin..."]:1: bad argument #-1 to 'dofile' (file not found)
stack traceback:
	[C]: in field 'dofile'
	[string " wesnoth.dofile("~add-ons/Legend_of_the_Invin..."]:1: in main chunk
```

The error message says that the argument to dofile wasn't found, but doesn't say what the value of the argument was, nor does it say the file name and line number where the error occurred. It should do one or the other to make debugging easier.

**Comment by gfgtdf** [comment]
It does give you the line me number (1) and the turncated string is exactly the name of the module where the error occured.

**Comment by jyrkive** [comment]
AFAIK, it's Lua that generates the error message, not our code.

This is the line that fetches the error message from the Lua stack: https://github.com/wesnoth/wesnoth/blob/318a6242ee855bca754404ad369d39069e4c594d/src/scripting/lua_kernel_base.cpp#L796

**Comment by jostephd** [comment]
> It does give you the line me number (1)

The file number is relative to the single-line lua code, not relative to a file, so it's hard to use it to find the code that raises the error.

> and the turncated string is exactly the name of the module where the error occured.

Yes, that's the whole point. The most important information, the _name of the file that produced the error_, is truncated. That's exactly the problem.

> AFAIK, it's Lua that generates the error message, not our code.

Okay, that's useful to know. The WML code that produced that error is:

```
_main.cfg-54-[lua]
_main.cfg-55-    code = <<
_main.cfg:56:	wesnoth.dofile("~add-ons/Legend_of_the_Invincibles/lua.lua")
_main.cfg:57:	wesnoth.dofile("~add-ons/Affably_Evil/wml_tags.lua")
_main.cfg-58-	>>
_main.cfg-59-[/lua]
```

so we could at least print "_main.cfg:55" before or after the lua error. I guess actually adding the filename to the error message would probably require changes in lua itself.

**Comment by gfgtdf** [comment]
> Yes, that's the whole point. The most important information, the name of the file that produced the error, is truncated. That's exactly the problem.

I think you misunderstood me here, the chunk where the error appeared ( the [Lua block, not the .Lua file) does not have a name, so Lua generates a dummy name by tutncating string content. So from the Lua perspective `" wesnoth.dofile("~add-ons/Legend_of_the_Invin..."` _is_ the filename from where that code comes from.


Ftr, in 1.15 you can give Lua chunk names by during [Lua] name=

**Comment by jyrkive** [comment]
The Lua chunk name is set in https://github.com/wesnoth/wesnoth/blob/318a6242ee855bca754404ad369d39069e4c594d/data/lua/wml-tags.lua#L271

What needs to be changed here is the fallback if `cfg.name` isn't set.

**Comment by jostephd** [comment]
No, @gfgtdf, I do understand that the truncated string is a filename and 1 is a line number relative to that anonymous file. It's just that, like I already said:

> it's hard to use it to find the code that raises the error.

What I want is a file name that exists on disk so I can open my editor on the line that raises the error message. Line numbers relative to anonymous files are a far cry from that.

> What needs to be changed here is the fallback if `cfg.name` isn't set.

Thanks for identifying that! I'll label this "good first issue" now since the WML-reading code already keeps track on line numbers (somehow, anyway, since WML syntax errors do have line numbers).

**Comment by gfgtdf** [comment]
This is totally not good first issue, the config objects have no information about line numbers, and this is not an error that happens during the ' wml reading'

**Comment by jyrkive** [comment]
> Thanks for identifying that!

I was already aware of it (I wrote our own custom implementation of Lua's `load()` function, and in the process I checked if we're using it in mainline).

@gfgtdf Agreed. I removed the label.
